### PR TITLE
Support Play Billing Library v4

### PR DESCRIPTION
**Pull request by syarihu**
This PR contains the diff for #14.

- Support Play Billing Library v4
    - https://developer.android.com/google/play/billing/release-notes?hl=en#4-0
    - Implement queryPurchasesAsync in DebugBillingClient
    - BillingClient.queryPurchases is deprecated, but leave it in place until it removes
    - We will not create a purchase screen for Multi-line subscriptions this time
    - Add getter and setter of ConnectionState to BillingStore for ConnectionState added since PBLv4

### CODE REVIEW

**Comment by syarihu** [comment]
@rharter Please review 🙇

File: library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt [hunk 1]
```diff
@@ -153,8 +148,9 @@ class DebugBillingClient(
 
   override fun launchBillingFlow(activity: Activity, params: BillingFlowParams): BillingResult {
     val intent = Intent(activity, DebugBillingActivity::class.java)
-    intent.putExtra(DebugBillingActivity.REQUEST_SKU_TYPE, params.skuType)
-    intent.putExtra(DebugBillingActivity.REQUEST_SKU, params.sku)
+    val skuDetails: List<SkuDetails> = params.zzj()
```
Comment (rharter) [inline library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt:151]: Is there a non-obfuscated param available to get this detail?  This value could easily change if the source BillingFlowParams file is changed in any way.

**Review by rharter — COMMENTED** [review]
Thanks, this is looking good, just added one question.

File: library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt (hunk 1 above)
Comment (syarihu) [reply library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt:151]: @rharter 
No. Unfortunately, only vrPurchaseFlow was not obfuscated in BillingFlowParams. This poor workaround is required, I agreed to your concern though.

Comment (syarihu) [reply library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt:151]: How about the following code?
```
    val skuDetails: List<SkuDetails> = params::class.java.declaredMethods
      .find {
        it.genericReturnType.toString() == "java.util.ArrayList<com.android.billingclient.api.SkuDetails>"
      }?.let { it.invoke(params) as ArrayList<SkuDetails> }
      .orEmpty()
```

Comment (rharter) [reply library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt:151]: I've filed an issue [here](https://issuetracker.google.com/issues/191995386), as obfuscating everything there seems like an oversight and a poor choice.

I think the reflection approach is the only viable one at this point, as any even point release will probably change the method name.

Comment (syarihu) [reply library/src/main/java/com/pixite/android/billingx/DebugBillingClient.kt:151]: Thanks for filed the issue. I agree with you, so I've starred the issue.
I will use the reflection approach at this time. And if the Play Billing Library is fixed, I’ll fix the BillingX as well. Thank you.

**Review by rharter — APPROVED** [review]

**Pull request merged**